src/plots/cartesian/axes.js

```diff
@@ -540,13 +541,43 @@ function autoShiftMonthBins(binStart, data, dtick, dataMin, calendar) {
 // Ticks and grids
 // ----------------------------------------------------
 
+// ensure we have minor tick0 and dtick calculated
+axes.prepMinorTicks = function(ax) {
+    var majorDtick = ax._majorDtick;
+    if(ax.tickmode === 'auto' || !ax.dtick) {
+        var nt = ax.nticks; // minor.nticks
+        var dist = majorDtick;
+
+        if(ax.type === 'date' && typeof majorDtick === 'string' && majorDtick.charAt(0) === 'M') {
+            var months = Number(majorDtick.substring(1));
+            dist = months * ONEAVGMONTH / (nt || 7);
+        } else if(ax.type === 'log') {
+            if(!nt) nt = 2;
+
+            if(nt > 1) {
+                if(typeof majorDtick === 'string' && majorDtick.charAt(0) === 'L') {
+                    ax.dtick = 'L' + (majorDtick.substring(1) / nt);
+                    return;
+                } else if(dist === 'D1') {
+                    dist = nt - 1;
+                } else if(dist === 'D2') {
+                    dist = nt === 2 ? 'D1' : nt;
+                } else {
+                    dist /= nt;
+                }
+            }
+        } else {
+            dist /= nt || 7;
```

Comment:
Small (minor? 😏 ) point: on linear axes, if I want `minor.dtick===dtick/10` I don't get this with `minor.nticks: 10`, I have to go to `11`

Also, I was expecting if I provided major `dtick` and `tick0` plus `minor.dtick`, that `minor.tick0` would be inherited from `tick0` - but it seems unless I explicitly provide it to match, `minor.tick0` reverts to the default.

![Screen Shot 2022-04-26 at 09 39 38](https://user-images.githubusercontent.com/2678795/165322491-0849f4f4-d562-4ddc-96ff-4f7c44d71933.png)